src/lib.rs

```diff
@@ -50,21 +50,43 @@ pub fn encrypt(remote_pub: &[u8], remote_auth: &[u8], salt: &[u8], data: &[u8])
 
 /// Decrypt a block using default AES128GCM encoding.
 ///
-/// param local_priv &str - The locally generated Private key as a raw, hex encoded string
+/// param components &str - The locally generated private key components.
 /// param auth &str - The locally generated auth token (this value was shared with the encryptor)
 /// param data &[u8] - The encrypted data block
 ///
-pub fn decrypt(local_priv: &[u8], auth: &[u8], data: &[u8]) -> Result<Vec<u8>> {
-    let priv_key = LocalKeyPairImpl::new(local_priv).unwrap();
+pub fn decrypt(components: &EcKeyComponents, auth: &[u8], data: &[u8]) -> Result<Vec<u8>> {
+    let priv_key = LocalKeyPairImpl::from_raw_components(components).unwrap();
     Aes128GcmEceWebPushImpl::decrypt(&priv_key, &auth, data)
 }
 
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn ec_point_from_d(d: &[u8]) -> Vec<u8> {
+        use openssl::{bn::*, ec::*, nid::Nid};
+        let group = EcGroup::from_curve_name(Nid::X9_62_PRIME256V1).unwrap();
+        let d = BigNum::from_slice(d).unwrap();
+        let mut bn_ctx = BigNumContext::new().unwrap();
+        let mut pub_key_point = EcPoint::new(&group).unwrap();
+        pub_key_point.mul_generator(&group, &d, &bn_ctx).unwrap();
+        pub_key_point
+            .to_bytes(&group, PointConversionForm::COMPRESSED, &mut bn_ctx)
+            .unwrap()
+    }
+
+    pub fn components_from_d(d: &[u8]) -> EcKeyComponents {
+        EcKeyComponents::new(d, &ec_point_from_d(d))
+    }
+}
```

Comment:
Ideally these tests would not depend on openssl-backend-specific behaviour, but we can address that in a followup.

Reply:
Yes it's a terrible hack sorry :/